waspc/data/Generator/templates/sdk/wasp/client/operations/rpc.ts

```diff
@@ -0,0 +1,38 @@
+import { type Route } from "wasp/client";
+import type {
+  _Awaited,
+  _ReturnType,
+} from "wasp/universal/types"
+
+// PRIVATE API (but should maybe be public, users define values of this type)
+export type Query<Input, Output> = QueryFunction<Input, Output> & QueryMetadata
+
+// PRIVATE API (but should maybe be public, users define values of this type)
+export type Action<Input, Output> = ClientOperation<Input, Output>
+
+// PRIVATE API
+export type QueryFunction<Input, Output> = ClientOperation<Input, Output>
+
+// PRIVATE API
+export type QueryMetadata = {
+  queryCacheKey: string[]
+  route: Route
+}
```

Comment:
I decided to expose both `route` and `queryCacheKey` to our users:
- `queryCacheKey` - Because they need it to unlock all the capabilities of tanstack-query (as our docs tell them)
- `route` - Because I didn't want to complicate things by introducing another "internal" type.

 Let me know if you think this is a problem

Reply:
Continuation of this whole discussion here: #2017